flash/vision/classification/data.py

```diff
@@ -13,369 +13,392 @@
+                "Transform should be a dict. "
+                f"Here are the available keys for your transforms: {DataPipeline.PREPROCESS_FUNCS}."
             )
+        return transform
+
+    @staticmethod
+    def default_train_transforms():
+        image_size = ImageClassificationData.image_size
+        if _KORNIA_AVAILABLE and not os.getenv("FLASH_TESTING", "0") == "1":
+            #  Better approach as all transforms are applied on tensor directly
+            return {
+                "post_tensor_transform": nn.Sequential(K.RandomResizedCrop(image_size), K.RandomHorizontalFlip()),
+                "per_batch_transform_on_device": nn.Sequential(
+                    K.Normalize(torch.tensor([0.485, 0.456, 0.406]), torch.tensor([0.229, 0.224, 0.225])),
+                )
+            }
         else:
-            valid_ds = (
-                FilepathDataset(
-                    filepaths=valid_filepaths,
-                    labels=valid_labels,
-                    loader=loader,
-                    transform=valid_transform,
-                ) if valid_filepaths else None
-            )
+            from torchvision import transforms as T  # noqa F811
+            return {
+                "pre_tensor_transform": nn.Sequential(T.RandomResizedCrop(image_size), T.RandomHorizontalFlip()),
+                "post_tensor_transform": T.Normalize([0.485, 0.456, 0.406], [0.229, 0.224, 0.225]),
+            }
+
+    @staticmethod
+    def default_valid_transforms():
+        image_size = ImageClassificationData.image_size
+        if _KORNIA_AVAILABLE and not os.getenv("FLASH_TESTING", "0") == "1":
+            #  Better approach as all transforms are applied on tensor directly
+            return {
+                "post_tensor_transform": nn.Sequential(K.RandomResizedCrop(image_size)),
+                "per_batch_transform_on_device": nn.Sequential(
+                    K.Normalize(torch.tensor([0.485, 0.456, 0.406]), torch.tensor([0.229, 0.224, 0.225])),
+                )
+            }
+        else:
+            from torchvision import transforms as T  # noqa F811
+            return {
+                "pre_tensor_transform": T.Compose([T.RandomResizedCrop(image_size)]),
+                "post_tensor_transform": T.Normalize([0.485, 0.456, 0.406], [0.229, 0.224, 0.225]),
+            }
 
-        test_ds = (
-            FilepathDataset(
-                filepaths=test_filepaths,
-                labels=test_labels,
-                loader=loader,
-                transform=valid_transform,
-            ) if test_filepaths else None
+    @property
+    def num_classes(self):
+        if self._num_classes is None:
+            if self._train_ds is not None:
+                self._num_classes = self._get_num_classes(self._train_ds)
+
+        return self._num_classes
+
+    def _get_num_classes(self, dataset: torch.utils.data.Dataset):
+        num_classes = self.get_dataset_attribute(dataset, "num_classes", None)
```

Comment:
maybe I am missing something here, but isn't this point to the self `num_classes` property? so going in a cycle as the property calls this method...

Reply:
If num_classes wasn't specified by the users, we infer it from the dataset.